fe/fe-core/src/main/java/com/starrocks/scheduler/MvTaskRunProcessor.java

```diff
@@ -0,0 +1,500 @@
+import com.google.common.collect.Range;
+import com.google.common.collect.Sets;
+import com.starrocks.analysis.AddPartitionClause;
+import com.starrocks.analysis.DistributionDesc;
+import com.starrocks.analysis.DropPartitionClause;
+import com.starrocks.analysis.Expr;
+import com.starrocks.analysis.FunctionCallExpr;
+import com.starrocks.analysis.HashDistributionDesc;
+import com.starrocks.analysis.InsertStmt;
+import com.starrocks.analysis.PartitionKeyDesc;
+import com.starrocks.analysis.PartitionNames;
+import com.starrocks.analysis.PartitionValue;
+import com.starrocks.analysis.QueryStmt;
+import com.starrocks.analysis.SingleRangePartitionDesc;
+import com.starrocks.analysis.SlotRef;
+import com.starrocks.analysis.StatementBase;
+import com.starrocks.catalog.Column;
+import com.starrocks.catalog.Database;
+import com.starrocks.catalog.ExpressionRangePartitionInfo;
+import com.starrocks.catalog.HashDistributionInfo;
+import com.starrocks.catalog.MaterializedView;
+import com.starrocks.catalog.OlapTable;
+import com.starrocks.catalog.Partition;
+import com.starrocks.catalog.PartitionInfo;
+import com.starrocks.catalog.PartitionKey;
+import com.starrocks.catalog.RangePartitionInfo;
+import com.starrocks.catalog.SinglePartitionInfo;
+import com.starrocks.common.Config;
+import com.starrocks.common.util.ExpressionPartitionUtil;
+import com.starrocks.metric.MetricRepo;
+import com.starrocks.metric.ResourceGroupMetricMgr;
+import com.starrocks.plugin.AuditEvent;
+import com.starrocks.proto.PQueryStatistics;
+import com.starrocks.qe.ConnectContext;
+import com.starrocks.qe.OriginStatement;
+import com.starrocks.qe.QueryState;
+import com.starrocks.qe.StmtExecutor;
+import com.starrocks.server.GlobalStateMgr;
+import com.starrocks.service.FrontendOptions;
+import com.starrocks.sql.analyzer.AST2SQL;
+import com.starrocks.sql.analyzer.AnalyzerUtils;
+import com.starrocks.sql.analyzer.SemanticException;
+import com.starrocks.sql.ast.QueryStatement;
+import com.starrocks.sql.ast.TableRelation;
+import com.starrocks.sql.common.SqlDigestBuilder;
+import com.starrocks.sql.parser.SqlParser;
+import org.apache.commons.codec.binary.Hex;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class MvTaskRunProcessor implements TaskRunProcessor {
+
+    private static final Logger LOG = LogManager.getLogger(MvTaskRunProcessor.class);
+
+    @Override
+    public void processTaskRun(TaskRunContext context) throws Exception {
+        Map<String, String> properties = context.getProperties();
+        // NOTE: need set mvId in Task's properties when creating
+        long mvId = Long.parseLong(properties.get("mvId"));
```

Comment:
would better add Preconditions check, if you sure the mvId must exist. besides, would better use constant variable for `mvId `